File: serverts/src/entities/Roles.ts
import { Column, Entity, ManyToMany, PrimaryGeneratedColumn} from "typeorm";
import { Users } from "./Users";

@Entity()
export class Roles{
    @PrimaryGeneratedColumn('uuid')
    id:string;

    @Column({length:50, unique:true})
    name:string;

    @ManyToMany(()=>Users, users=>users.roles, {onDelete:'SET NULL'})
    users:Users[];    

}

File: serverjava/src/main/java/com/proyecto/serverjava/entities/Roles.java
package com.proyecto.serverjava.entities;

import java.util.List;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "roles")
@Data
public class Roles {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(length = 50, unique = true)
    private String name;

    @ManyToMany(mappedBy = "roles")
    @JsonIgnoreProperties({"roles"})
    private List<Users> users;
}
